File: app/src/main/java/com/pirateman/exercisemanager/selectedexercise/SelectedExerciseHolder.java
package com.pirateman.exercisemanager.selectedexercise;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.widget.Toast;

import com.pirateman.exercisemanager.databinding.SelectedExerciseItemBinding;
import com.pirateman.exercisemanager.exercise.BaseHolder;
import com.pirateman.exercisemanager.exercise.Exercise;
import com.pirateman.exercisemanager.interval.IntervalActivity;

public class SelectedExerciseHolder extends BaseHolder implements View.OnClickListener, View.OnLongClickListener
{
    private SelectedExerciseItemBinding binding;

    public SelectedExerciseHolder(SelectedExerciseItemBinding binding, final Context context)
    {
        super(binding, context);
        this.binding = binding;
        this.context = context;

        binding.getRoot().setOnClickListener(this);
        binding.getRoot().setOnLongClickListener(this);
    }

    public void bind(Exercise exercise)
    {
        binding.setExerciseRecord(exercise);
    }

    @Override
    public void onClick(View v) {

        Intent intent = new Intent(context, IntervalActivity.class);
        intent.putExtra("exercise", binding.getExerciseRecord());
        context.startActivity(intent);
    }

    @Override
    public boolean onLongClick(View v) {
        Toast.makeText(context, "Go to next activity", Toast.LENGTH_SHORT).show();

        return true;
    }
}

File: app/src/main/java/com/pirateman/exercisemanager/selectedexercise/SelectedExerciseHolder.kt
package com.pirateman.exercisemanager.selectedexercise

import android.os.Build
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import com.pirateman.exercisemanager.databinding.RecyclerViewSelectedExerciseItemBinding

@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
class SelectedExerciseHolder(private val viewBinding: RecyclerViewSelectedExerciseItemBinding) : RecyclerView.ViewHolder(viewBinding.root) {

    //For potential future use
    val tvSelectedExerciseName = viewBinding.tvSelectedExerciseName
    val imgDeleteExercise = viewBinding.imgDelete

    init {
        viewBinding.root.isClickable = true

//        imgDeleteExercise.setOnClickListener {
//            (itemView as ViewGroup).removeView(view)
//        }

    }



//    @RequiresApi(Build.VERSION_CODES.M)
//    override fun onClick(itemView: View) {
//
//        val ex = itemView.tag as Exercise
//        //exerciseViewModel.currentExercise = ex
//
//        if (itemView == imgDeleteExercise) {
//            //adapterPosition
////            if (itemCount > 0) {
//
//                //exerciseViewModel.setSelectedExercise(ex, false)
//
//                //Id should be the same as the position in the recyclerview
//
////                exerciseList.removeAt(ex.id)
////                notifyItemRemoved(ex.id)
////            }
//        }
//        else{
//            Navigation.findNavController(itemView).navigate(R.id.workoutFragment)
//        }
//
//
//
////
////        if (itemView.equals(imgDeleteExercise)){
////
////        }
//
//        //val context = itemView.context
//
//    }



}